### PR TITLE
Update docs for typescript sdk for browser

### PR DESCRIPTION
**Pull request by kevinpagtakhan**
# Dev Docs PR

## Description

Update Typescript SDK for Browser to include more complete and accurate information.

## Deadline

🤔 

## Change type

- [ ] Doc bug fix. Fixes #[insert issue number]. Amplitude contributors include Jira issue number. 
- [x] Doc update.
- [ ] New documentation.
- [ ] Non-documentation related fix or update.

# PR checklist:

- [x] My documentation follows the style guidelines of this project.
- [x] I previewed my documentation on a local server using `mkdocs serve`.
- [x] Running `mkdocs serve` didn't generate any failures.
- [x] I have performed a self-review of my own documentation.

@amplitude-dev-docs
@caseyamp

### CODE REVIEW

**Comment by kevinpagtakhan** [comment]
This is the page the PR modified: https://pr-117.d19s7xzcva2mw3.amplifyapp.com/data/sdks/typescript-browser/#sdks

File: docs/data/sdks/typescript-browser.md [hunk 1]
```diff
@@ -99,42 +92,137 @@ track('Button Clicked', eventProperties);
...
+identifyObj.remove('unique-locations', 'JFK')
+
+identify(identifyObj);
+```
 
 ### User groups
 
 --8<-- "includes/editions-growth-enterprise-with-accounts.md"
 
+Amplitude supports assigning users to groups and performing queries such as Count by Distinct on those groups. An example would be if you want to group your users based on what organization they are in by using an 'orgId'. You can designate Joe to be in 'orgId' '10' while Sue is in 'orgId' '15'. When performing a query in our Event Segmentation chart, you can then select "..performed by" 'orgId' to query the number of different organizations that have performed a specific event. As long as at least one member of that group has performed the specific event, that group will be included in the count.
```
Comment (caseyamp) [inline docs/data/sdks/typescript-browser.md:222]:
```suggestion
--8<-- "includes/groups-intro-paragraph.md"
```

We have a reusable that contains all this.

File: docs/data/sdks/typescript-browser.md [hunk 2]
```diff
@@ -99,42 +92,137 @@ track('Button Clicked', eventProperties);
...
+
```
Comment (caseyamp) [inline docs/data/sdks/typescript-browser.md:223]:
```suggestion
```

File: docs/data/sdks/typescript-browser.md [hunk 3]
```diff
@@ -99,42 +92,137 @@ track('Button Clicked', eventProperties);
...
+When setting groups, you will need to define a groupType and groupName(s). In the above example, 'orgId' is the groupType and the values '10' and '15' are groupName(s). Another example of a groupType could be 'sport' with groupName(s) like 'tennis' and 'baseball'. You can use setGroup() to designate which groups a user belongs to. Note: This will also set the 'groupType:groupName' as a user property. This will overwrite any existing groupName value set for that user's groupType, as well as the corresponding user property value. groupType is a string and groupName can be either a string or an array of strings to indicate a user being in multiple groups (for example, if Joe is in 'orgId' '10' and '16', then the groupName would be '[10, 16]'). Here is what your code might look like.
```
Comment (caseyamp) [inline docs/data/sdks/typescript-browser.md:224]:
```suggestion
```

File: docs/data/sdks/typescript-browser.md [hunk 4]
```diff
@@ -180,6 +269,18 @@ const event = new Revenue()
 revenue(event);
 ```
 
+#### Revenue interface
+
+Name |  Type  |Description | Default
+-----|-------|--------------|--------
+product_id (optional) | string | An identifier for the product. We recommend something like the Google Play Store product ID. | null
```
Comment (caseyamp) [inline docs/data/sdks/typescript-browser.md:276]:
```suggestion
|`product_id` | Optional. String. An identifier for the product. We recommend something like the Google Play Store product ID. Defaults to null. |
```

File: docs/data/sdks/typescript-browser.md [hunk 5]
```diff
@@ -180,6 +269,18 @@ const event = new Revenue()
 revenue(event);
 ```
 
+#### Revenue interface
+
+Name |  Type  |Description | Default
```
Comment (caseyamp) [inline docs/data/sdks/typescript-browser.md:274]:
```suggestion
|Name | Description |
```

File: docs/data/sdks/typescript-browser.md [hunk 6]
```diff
@@ -180,6 +269,18 @@ const event = new Revenue()
...
+-----|-------|--------------|--------
```
Comment (caseyamp) [inline docs/data/sdks/typescript-browser.md:275]:
```suggestion
|-----|-------|
```

File: docs/data/sdks/typescript-browser.md [hunk 7]
```diff
@@ -180,6 +269,18 @@ const event = new Revenue()
...
+product_id (optional) | string | An identifier for the product. We recommend something like the Google Play Store product ID. | null
+quantity *(required)* | int| The quantity of products purchased. Note: revenue = quantity * price | 1
```
Comment (caseyamp) [inline docs/data/sdks/typescript-browser.md:277]:
```suggestion
|`quantity` | Required. Int. The quantity of products purchased. Note: revenue = quantity * price. Defaults to 1|
```

File: docs/data/sdks/typescript-browser.md [hunk 8]
```diff
@@ -180,6 +269,18 @@ const event = new Revenue()
...
+price *(required)* | Double | The price of the products purchased, and this can be negative. Note: revenue = quantity * price | null
+revenue_type (optional, *required for revenue verification*) | String| The type of revenue (e.g. tax, refund, income). | null
```
Comment (caseyamp) [inline docs/data/sdks/typescript-browser.md:279]:
```suggestion
|`revenue_type` | Optional, but required for revenue verification. String. The type of revenue (e.g. tax, refund, income).  Defaults to null.|
```

File: docs/data/sdks/typescript-browser.md [hunk 9]
```diff
@@ -180,6 +269,18 @@ const event = new Revenue()
...
+receipt (optional) | String| The receipt identifier of the revenue. | null
```
Comment (caseyamp) [inline docs/data/sdks/typescript-browser.md:280]:
```suggestion
|`receipt`| Optional. String. The receipt identifier of the revenue. Defaults to null|
```

File: docs/data/sdks/typescript-browser.md [hunk 10]
```diff
@@ -180,6 +269,18 @@ const event = new Revenue()
...
+receipt_sig (optional, *required for revenue verification*) | String| The receipt signature of the revenue. | null
+properties (optional) | JSONObject| An object of event properties to include in the revenue event.| null
```
Comment (caseyamp) [inline docs/data/sdks/typescript-browser.md:282]:
```suggestion
|`properties`| Optional. JSONObject. An object of event properties to include in the revenue event. Defaults to null.
```

**Review by caseyamp — CHANGES_REQUESTED** [review]
Great work. @kevinpagtakhan -- I made a few changes to follow our formatting standards. After you accept them you're free to merge!

**Comment by kevinpagtakhan** [comment]
@caseyamp i have committed the changes you suggested but the preview check isn't completing and I'm unable to merge. can you take a look please? thank you

File: docs/data/sdks/typescript-browser.md [hunk 11]
```diff
@@ -99,42 +92,135 @@ track('Button Clicked', eventProperties);
 
 User properties help you understand your users at the time they performed some action within your app such as their device details, their preferences, or language.
 
+Identify is for setting the user properties of a particular user without sending any event. The SDK supports the operations `set`, `setOnce`, `unset`, `add`, `append`, `prepend`, `preInsert`, `postInsert`, and `remove` on individual user properties. The operations are declared via a provided Identify interface. Multiple operations can be chained together in a single Identify object. The Identify object is then passed to the Amplitude client to send to the server.
+
+!!!warning "SDK Resources"
```
Comment (caseyamp) [inline docs/data/sdks/typescript-browser.md:97]:
```suggestion
!!!note 
```

File: docs/data/sdks/typescript-browser.md [hunk 12]
```diff
@@ -385,4 +389,49 @@ track('Button Clicked').promise.then((result) => {
...
+
+This method contains logic for preparing the plugin for use and has config as a parameter. The expected return value is undefined. A typical use for this method, is to copy configuration from config or instantiate plugin dependencies. This method is called when the plugin is registered to the client via `client.add()`.
+
+#### Plugin.execute
+
+This method contains the logic for processing events and has event as parameter. If used as enrichment type plugin, the expected return value is the modified/enriched event; while if used as a destination type plugin, the expected return value is undefined. This method is called for each event, including Identify, GroupIdentify and Revenue events, that is instrumented using the client interface.
+
+#### Plugin examples
+
+#### Enrichment type plugin
```
Comment (caseyamp) [inline docs/data/sdks/typescript-browser.md:406]:
```suggestion
##### Enrichment type plugin
```

**Review by caseyamp — APPROVED** [review]

**Pull request merged**